src/overlays/actors/ovl_En_Tanron3/z_en_tanron3.c

```diff
@@ -28,39 +36,419 @@ const ActorInit En_Tanron3_InitVars = {
+    this->actor.speedXZ = 5.0f;
+    this->speedMaxStep = 0.5f;
+    this->deviation.x = randPlusMinusPoint5Scaled(500.0f);
+    this->deviation.y = randPlusMinusPoint5Scaled(100.0f);
+    this->deviation.z = randPlusMinusPoint5Scaled(500.0f);
+    Math_Vec3f_Copy(&this->targetPos, &this->actor.world.pos);
+    this->timer = Rand_ZeroFloat(100.0f);
+}
+
+/**
+ * This controls the vast majority of the fish's behavior while it's alive, including:
+ * - deciding whether to be hostile or not
+ * - determing whether the fish is beached or not
+ * - swimming towards the player to attack them
+ * - swimming around idly if the player is out of range
+ * - flopping around on land if it beaches itself
+ */
+void EnTanron3_Live(EnTanron3* this, GlobalContext* globalCtx) {
+    s32 atanTemp;
+    f32 xDistance;
+    f32 yDistance;
+    f32 zDistance;
+    f32 xzDistance;
+    f32 extraScaleY = 0.0f;
+    Player* player = GET_PLAYER(globalCtx);
+
+    this->skelAnime.curFrame = 4.0f;
+    if ((player->actor.bgCheckFlags & 1) && player->actor.shape.feetPos[0].y >= 438.0f) {
+        // Player is standing on the central platform, so stop chasing them
+        this->isNonHostile = true;
+    } else if (this->isNonHostile && this->workTimer[TANRON3_WORK_TIMER_WAIT] == 0 && !(this->timer & 0x1F)) {
+        xDistance = this->targetPos.x - player->actor.world.pos.x;
+        zDistance = this->targetPos.z - player->actor.world.pos.z;
+        if (sqrtf(SQ(xDistance) + SQ(zDistance)) < 500.0f) {
+            // Player is in the water and close enough, so start chasing them
+            this->isNonHostile = false;
+            this->workTimer[TANRON3_WORK_TIMER_ATTACK] = 150;
+        }
+    }
+
+    if (this->actor.world.pos.y < this->waterSurfaceYPos) {
+        // The fish is below the water's surface, so it's no longer beached if it was before
+        this->isBeached = false;
+        switch (this->isNonHostile) {
+            case false:
+                this->targetSpeedXZ = 5.0f;
+                this->targetRotationStep = 0x1000;
+                this->nextRotationAngle = 0x3A98;
+
+                Math_Vec3f_Copy(&this->targetPos, &player->actor.world.pos);
+                if (!(this->timer & 0xF)) {
+                    if (Rand_ZeroOne() < 0.5f && this->actor.xzDistToPlayer <= 200.0f) {
+                        Audio_PlayActorSound2(&this->actor, NA_SE_EN_PIRANHA_ATTACK);
+                    }
+                }
+
+                // If the player gets eaten by Gyorg, or if the attack timer ran out,
+                // stop chasing the player for a little bit.
+                if (this->workTimer[TANRON3_WORK_TIMER_ATTACK] == 0 || (player->stateFlags2 & 0x80)) {
+                    this->workTimer[TANRON3_WORK_TIMER_WAIT] = 150;
+                    this->isNonHostile = true;
+                }
+                break;
+            case true:
+                if (sGyorg->unk_324 != 0 && !(this->timer & 0x7)) {
```

Comment:
Just to be sure, this bitwise and can't be changed to a `% 8`, right?

Reply:
I feel like @engineer124 already asked about all of them. And I wouldn't want them inconsistent in the file anyway.

Reply:
Yeah, they did. However, there is one inconsistency where a `%` worked, so let me remove that.